### PR TITLE
Fix: Pages are not being scanned by Language Processing

### LINKED ISSUE #161: Pages are not being scanned by Language Processing
**Describe the bug**
Pages content is being scanned by Watson even though the settings indicate that they should be.

**Steps to Reproduce**

1. Enable Pages to be scanned in the Language Settings panel
2. Create a new Page and add content.
3. Publish the Page

**Expected behavior**
We should see Watson Keywords, Entities, Categories and Concepts added to the pages.

### PR DESCRIPTION
**Pull request by dinhtungdu**
### Description of the Change

- Add page to rest route check (`is_rest_route`).
- Add a filter (`classifai_rest_bases`) to support custom post type with custom rest base.
- Remove unused enqueue.
- Remove deprecated (and removed) function (`is_gutenberg_page`).

### Benefits

Pages now are scanned and tagged by Language Processing.

### Verification Process

1. Edit a page then hit the Update button.
2. See the page updated normally without any error.
3. Go back to the pages list, see the page processed by Waston.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/10up/classifai/blob/develop/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Applicable Issues
Closes #161

### CODE REVIEW

File: includes/Classifai/Admin/SavePostHandler.php [hunk 1]
```diff
@@ -188,9 +188,20 @@ public function show_error_if() {
...
 	 */
 	public function is_rest_route() {
-		if ( isset( $_SERVER['REQUEST_URI'] ) && false !== strpos( $_SERVER['REQUEST_URI'], 'wp-json/wp/v2/post' ) ) {
-			return true;
+
+		if ( ! isset( $_SERVER['REQUEST_URI'] ) ) {
+			return false;
 		}
+
+		// Support custome post types with custom rest base.
```
Comment (ryanwelcher) [inline includes/Classifai/Admin/SavePostHandler.php]:
```suggestion
		// Support custom post types with custom rest base.
```

**Comment by ryanwelcher** [comment]
@dinhtungdu great work on this! I'm thinking we should have some unit tests to cover pages - any objections?

**Comment by ryanwelcher** [comment]
@dinhtungdu I'm not sure what we're testing here. I think we want to be sure that when a page is saved, it is being scanned.

**Comment by dinhtungdu** [comment]
@ryanwelcher I added test for `is_rest_route` only because it gets updated in this PR and it's the cause of the issue.

The issue only happens for block editor only, so I think WPAcceptance tests will be better fit and more straight forward. We can wait for #179 to be merged then continue work on this?

**Pull request merged**